### PR TITLE
KAFKA-6166: Update javadoc for streams config

### PR DESCRIPTION
**Pull request by guozhangwang**
This PR piggy-back a few different javadoc changes:

1. KIP-220: add admin client prefix.
2. Clarify on user customized configs.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by guozhangwang** [comment]
@h314to @wushujames for reviews.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 1]
```diff
@@ -78,8 +79,26 @@
...
+ * The consumer/producer/admin prefix can also be used to distinguish these custom config values passed to different clients with the same config name.
+ * * Example:
+ * <pre>{@code
+ * Properties streamsProperties = new Properties();
+ * // sets "my.custom.config" to "foo" for consumer only
+ * streamsProperties.put(StreamsConfig.consumerPrefix("my.custom.config"), "foo");
+ * // sets "my.custom.config" to "bar" for producer only
+ * streamsProperties.put(StreamsConfig.producerPrefix("my.custom.config"), "bar");
+ * // sets "my.custom.config2" to "boom" for all clients universally
+ * streamsProperties.put("my.custom.config", "boom");
```
Comment (wushujames) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: Typo. `my.custom.config` -> `my.custom.config2`

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java [hunk 2]
```diff
@@ -56,9 +57,9 @@
 
 /**
  * Configuration for a {@link KafkaStreams} instance.
- * Can also be used to configure the Kafka Streams internal {@link KafkaConsumer} and {@link KafkaProducer}.
- * To avoid consumer/producer property conflicts, you should prefix those properties using
- * {@link #consumerPrefix(String)} and {@link #producerPrefix(String)}, respectively.
+ * Can also be used to configure the Kafka Streams internal {@link KafkaConsumer}, {@link KafkaProducer} and {@link AdminClient}.
+ * To avoid consumer/producer/admin property conflicts, you should prefix those properties using
+ * {@link #consumerPrefix(String)}, {@link #producerPrefix(String)} and {@link #adminClientPrefix(String)}, respectively.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java:62]: nit: `, and` (missing comma)

**Review by mjsax — APPROVED** [review]
Feel free to merge after addressing both comments.

File: streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java (hunk 1 above)
Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java]: ack.

**Pull request merged**